projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts

```diff
@@ -124,3 +148,44 @@ describe('ReportBPComponent', () => {
   });
 
 });
+
+describe('snackbar test', () => {
+  let fixture: ComponentFixture<ReportBPComponent>;
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule, NoopAnimationsModule, MatInputModule, MatCardModule],
+      declarations: [ReportBPComponent],
+      providers: [ MatSnackBar],
+    }).compileComponents();
```

Comment:
This testing module is also using real FhirDataService. If a test were to call onSubmit instead of open, it would try to create data on the FHIR server.

Reply:
Okay, @semicolin  I will use FhirDataServiceSpy instead of the real FhirDataService.